library/core/src/ptr/const_ptr.rs

```diff
@@ -799,6 +807,108 @@ impl<T: ?Sized> *const T {
+    /// runtime, and may be exploited by optimizations. You can use `(self as usize).sub(origin as
+    /// usize) / mem::size_of::<T>()` to avoid this requirement.
+    ///
+    /// [allocated object]: crate::ptr#allocated-object
+    ///
+    /// # Panics
+    ///
+    /// This function panics if `T` is a Zero-Sized Type ("ZST").
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// #![feature(ptr_wrapping_offset_from)]
+    /// let a = [0; 2];
+    /// let ptr1: *const i32 = &a[1];
+    /// let ptr2: *const i32 = a.as_ptr().wrapping_offset(3); // out-of-bounds!
+    /// unsafe {
+    ///     assert_eq!(ptr2.wrapping_offset_from(ptr1), 2);
+    ///     assert_eq!(ptr1.wrapping_offset_from(ptr2), -2);
+    ///     assert_eq!(ptr1.wrapping_offset(2), ptr2);
+    ///     assert_eq!(ptr2.wrapping_offset(-2), ptr1);
+    /// }
+    /// ```
+    ///
+    /// *Incorrect* usage:
+    ///
+    /// ```rust,no_run
+    /// #![feature(ptr_wrapping_offset_from)]
+    /// let ptr1 = Box::into_raw(Box::new(0u8)) as *const u8;
+    /// let ptr2 = Box::into_raw(Box::new(1u8)) as *const u8;
+    /// let diff = (ptr2 as isize).wrapping_sub(ptr1 as isize);
+    /// // Make ptr2_other an "alias" of ptr2, but derived from ptr1.
+    /// let ptr2_other = (ptr1 as *const u8).wrapping_offset(diff);
+    /// assert_eq!(ptr2 as usize, ptr2_other as usize);
+    /// // Since ptr2_other and ptr2 are derived from pointers to different objects,
+    /// // computing their offset is undefined behavior, even though
+    /// // they point to the same address!
+    /// unsafe {
+    ///     let zero = ptr2_other.wrapping_offset_from(ptr2); // Undefined Behavior
+    /// }
+    /// ```
+    #[unstable(feature = "ptr_wrapping_offset_from", issue = "none")]
+    #[rustc_const_unstable(feature = "ptr_wrapping_offset_from", issue = "none")]
+    #[inline]
+    #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
+    #[cfg(not(bootstrap))]
+    pub const unsafe fn wrapping_offset_from(self, origin: *const T) -> isize
+    where
+        T: Sized,
+    {
+        // SAFETY: the caller must uphold the safety contract for `ptr_offset_from`.
+        unsafe { intrinsics::ptr_wrapping_offset_from(self, origin) }
+    }
+
+    /// Calculates the distance between two pointers using wrapping arithmetic. The returned value
+    /// is in units of **bytes**.
+    ///
+    /// This is purely a convenience for casting to a `u8` pointer and using
+    /// [`wrapping_offset_from`][pointer::wrapping_offset_from] on it. See that method for
+    /// documentation and safety requirements.
+    ///
+    /// For non-`Sized` pointees this operation considers only the data pointers,
+    /// ignoring the metadata.
+    #[inline(always)]
+    #[unstable(feature = "ptr_wrapping_offset_from", issue = "none")]
+    #[rustc_const_unstable(feature = "ptr_wrapping_offset_from", issue = "none")]
```

Comment:
This could be tracked with `wrapping_offset_from` or with the other `bytes` methods, not sure which is more appropriate.